src/components/structures/RoomView.js

```diff
@@ -1412,6 +1260,33 @@ module.exports = React.createClass({
                     </div>
             }
 
+
+            // if we have search results, we keep the messagepanel (so that it preserves its
+            // scroll state), but hide it.
+            var searchResultsPanel;
+            var hideMessagePanel = false;
+
+            if (this.state.searchResults) {
+                searchResultsPanel = (
+                    <ScrollPanel ref="searchResultsPanel" className="mx_RoomView_messagePanel"
+                            onFillRequest={ this.onSearchResultsFillRequest }>
+                        <li className={scrollheader_classes}></li>
+                        {this.getSearchResultTiles()}
+                    </ScrollPanel>
+                );
+                hideMessagePanel = true;
+            }
+
+            var messagePanel = (
```

Comment:
Well, the point is to allow it to maintain its state - in particular, the scroll position. We *could* do that by remembering to save the scroll position we switch views, but it's much easier to keep the object around in the DOM.

The react docs seem to suggest this is the way to achieve it: http://facebook.github.io/react/docs/multiple-components.html#stateful-children